Add support for ssh algorithms, compaitibale with old embedded devices

xiaoyun94 opened the pull request:
Add support for ssh algorithms, compaitibale with old embeded devices

Commit:
Add support for ssh algorithms, compaitibale with old embeded devices

GitMensch commented:
Sounds good - can you please add something to the readme for this setting?
Should we also [update SSH2](https://github.com/mscdex/ssh2/compare/v1.6.0...v1.15.0) along? This seems to add [optional strict key support](https://github.com/mscdex/ssh2/commit/97b223f8891b96d6fc054df5ab1d5a1a545da2a3).

xiaoyun94 commented:
> Sounds good - can you please add something to the readme for this setting? Should we also [update SSH2](https://github.com/mscdex/ssh2/compare/v1.6.0...v1.15.0) along? This seems to add [optional strict key support](https://github.com/mscdex/ssh2/commit/97b223f8891b96d6fc054df5ab1d5a1a545da2a3).



> Sounds good - can you please add something to the readme for this setting? Should we also [update SSH2](https://github.com/mscdex/ssh2/compare/v1.6.0...v1.15.0) along? This seems to add [optional strict key support](https://github.com/mscdex/ssh2/commit/97b223f8891b96d6fc054df5ab1d5a1a545da2a3).

These parameters i choosed are listed in README.md of `mscdex/ssh`

GitMensch commented on README.md:
```diff
@@ -151,6 +155,8 @@ For X11 forwarding to work you first need to enable it in your Display Manager a
 connections. To allow connections you can either add an entry for applications or run `xhost +`
 in the console while you are debugging and turn it off again when you are done using `xhost -`.
 
+SSH algorithms that some old embeded devices use may be outdate, there is a compiable method via using `algorithms`. `kex`, `cipher`,`compress`, `hmac`, `serverHostKey` are known to be supported in `algorithms`, data format of these keys is array. Specific content can be referred to [msc/ssh](https://github.com/mscdex/ssh2) (`Client methods`->`connect`->`algorithms`) .
```
I suggest the following text adjustment:

> SSH algorithms used by some old embedded devices may be out of date, there is a compileable method using `algorithms`, `kex`, `cipher`,`compress`, `hmac`, `serverHostKey` are known to be supported in `algorithms`. The data format of these keys is array. Supported values can be found in [mscdex/ssh2](https://github.com/mscdex/ssh2/blob/master/README.md#client-methods)(`Client methods`->`connect`->`algorithms`).
which includes a change to the deeplink https://github.com/mscdex/ssh2/blob/master/README.md#client-methods .

xiaoyun94 replied:
done

GitMensch commented on README.md:
````diff
@@ -122,7 +122,7 @@ Debugging using ssh automatically converts all paths between client & server and
 Simply add a `ssh` object in your `launch` request.
 
-```
+```JSON
````
use `jsonc` to fix the highlighting

xiaoyun94 replied:
ok

GitMensch reviewed (changes requested):
LGTM.

Do you want to add  your change to the CHANGELOG (possibly along with further REAMDE adjustments) or should I pull and do another commit?

xiaoyun94 commented:
> CHANGELOG

I suggest you do that. I am not good at English writing😂

xiaoyun94 commented:
> LGTM.
> 
> Do you want to add your change to the CHANGELOG (possibly along with further REAMDE adjustments) or should I pull and do another commit?

I suggest you do that. I am not good at English writing😂

xiaoyun94 commented:
@GitMensch what should i do next

GitMensch commented:
Can you try to update to ssh2 1.15 and retest connecting to your old embedded device please?

GitMensch commented:
For the ChangeLog you can use

> -add support for ssh algorithms, compatible with old embedded devices ([@xiaoyun94])

and for the first PR add your entry below "<!-- Contributors listed alphabetically -->" as well, please.

xiaoyun94 commented:
> For the ChangeLog you can use
> 
> > -add support for ssh algorithms, compatible with old embedded devices ([@xiaoyun94])
> 
> and for the first PR add your entry below "" as well, please.

sorry, i have no deviced board to test recently

xiaoyun94 commented:
> below

no device available recently, so sorry

xiaoyun94 commented:
> Can you try to update to ssh2 1.15 and retest connecting to your old embedded device please?

no device available recently, so sorry

xiaoyun94 committed:
Update gdb_expansion.ts

Resolved the issue that the input parameter value of a function cannot be displayed when the structure pointer is used as the input parameter of the function.
